.github/workflows/changes.yml

```diff
@@ -0,0 +1,334 @@
+      clickhouse: ${{ steps.filter.outputs.clickhouse }}
+      databend: ${{ steps.filter.outputs.databend }}
+      datadog: ${{ steps.filter.outputs.datadog }}
+      dnstap: ${{ steps.filter.outputs.dnstap }}
+      docker-logs: ${{ steps.filter.outputs.docker-logs }}
+      elasticsearch: ${{ steps.filter.outputs.elasticsearch }}
+      eventstoredb: ${{ steps.filter.outputs.eventstoredb }}
+      fluent: ${{ steps.filter.outputs.fluent }}
+      gcp: ${{ steps.filter.outputs.gcp }}
+      humio: ${{ steps.filter.outputs.humio }}
+      http-client: ${{ steps.filter.outputs.http-client }}
+      influxdb: ${{ steps.filter.outputs.influxdb }}
+      kafka: ${{ steps.filter.outputs.kafka }}
+      logstash: ${{ steps.filter.outputs.logstash }}
+      loki: ${{ steps.filter.outputs.loki }}
+      mongodb: ${{ steps.filter.outputs.mongodb }}
+      nats: ${{ steps.filter.outputs.nats }}
+      nginx: ${{ steps.filter.outputs.nginx }}
+      opentelemetry: ${{ steps.filter.outputs.opentelemetry }}
+      postgres: ${{ steps.filter.outputs.postgres }}
+      prometheus: ${{ steps.filter.outputs.prometheus }}
+      pulsar: ${{ steps.filter.outputs.pulsar }}
+      redis: ${{ steps.filter.outputs.redis }}
+      shutdown: ${{ steps.filter.outputs.shutdown }}
+      splunk: ${{ steps.filter.outputs.splunk }}
+      webhdfs: ${{ steps.filter.outputs.webhdfs }}
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        base: ${{ inputs.base_ref }}
+        ref: ${{ inputs.head_ref }}
+        filters: |
+          source:
+            - ".github/workflows/test.yml"
+            - ".cargo/**"
+            - "benches/**"
+            - "lib/**"
+            - "proto/**"
+            - "scripts/**"
+            - "src/**"
+            - "tests/**"
+            - "build.rs"
+            - "Cargo.lock"
+            - "Cargo.toml"
+            - "Makefile"
+            - "rust-toolchain.toml"
+            - "vdev/**"
+          deny:
+            - 'deny.toml'
+            - "vdev/**"
+          dependencies:
+            - ".cargo/**"
+            - 'Cargo.toml'
+            - 'Cargo.lock'
+            - 'rust-toolchain.toml'
+            - '.github/workflows/pr.yml'
+            - 'Makefile'
+            - 'scripts/cross/**'
+            - "vdev/**"
+          cue:
+            - 'website/cue/**'
+            - "vdev"
+          component_docs:
+            - 'scripts/generate-component-docs.rb'
+            - "vdev/**"
+          markdown:
+            - '**/**.md'
+            - "vdev/**"
+          internal_events:
+            - 'src/internal_events/**'
+            - "vdev/**"
+          docker:
+            - 'distribution/docker/**'
+            - "vdev/**"
+          install:
+            - ".github/workflows/install-sh.yml"
+            - "distribution/install.sh"
+          k8s:
+            - "src/sources/kubernetes_logs/**"
+          amqp:
```

Comment:
For the integration tests, it struck me that we might want to check for changes in common places like 'lib/' or something like that, and run all of them.
Wanted to get team's thoughts on that as I'm sure there would be some cases where that wouldn't be necessary, but figured "core" code utilized everywhere might benefit from that.